File: src/com/colacelli/ircbot/plugins/help/PluginHelp.java
package com.colacelli.ircbot.plugins.help;

public class PluginHelp {
    protected String command;
    protected int access;
    protected String help;
    protected String[] args;

    public PluginHelp(String command, int access, String help, String... args) {
        this.command = command;
        this.access = access;
        this.help = help;
        this.args = args;
    }

    public PluginHelp(String command, String help, String... args) {
        this.command = command;
        this.access = 0;
        this.help = help;
        this.args = args;
    }

    public int getAccess() {
        return access;
    }

    public String getCommand() {
        return command;
    }

    @Override
    public String toString() {
        StringBuilder text = new StringBuilder();
        text.append(command);

        for (String arg : args) {
            text.append(" <");
            text.append(arg);
            text.append(">");
        }

        text.append(": ");
        text.append(help);

        return text.toString();
    }
}

File: src/com/colacelli/ircbot/plugins/help/PluginHelp.kt
package com.colacelli.ircbot.plugins.help

import com.colacelli.ircbot.plugins.access.IRCBotAccess

open class PluginHelp(val command : String, val access : IRCBotAccess.Level, val help : String, vararg val args : String) {
    override fun toString() : String {
        var text = command

        args.forEach {
            text += " <$it>"
        }

        text += ": $help"

        return text
    }
}